### PR TITLE
[Fluid] Correcting DistanceSmoothingProcess

### PR DESCRIPTION
**Pull request by mrhashemi**
**Description**
The test is revised and checked. Moreover, some small fixes are implemented.

Please mark the PR with appropriate tags: 
- Test
- Small Fixes

**Changelog**
Please summarize the changes in one list to generate the changelog:

### CODE REVIEW

**Comment by rubenzorrilla** [comment]
Which was the problem?

**Comment by mrhashemi** [comment]
> Which was the problem?

I do not know the exact reason but the test was failed, e.g. https://github.com/KratosMultiphysics/Kratos/pull/7738#issuecomment-721816316
So, I checked the process for 2D/3D cases on the **updated** master.
 
Moreover, I took the advantage of this PR to separate minor corrections of SmoothingProcess from https://github.com/KratosMultiphysics/Kratos/pull/7394

File: applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp [hunk 1]
```diff
@@ -122,7 +120,9 @@ void DistanceSmoothingProcess<TDim, TSparseSpace, TDenseSpace, TLinearSolver>::E
 
             auto& n_nodes = rNode.GetValue(NEIGHBOUR_NODES);
             for (unsigned int j = 0; j < n_nodes.size(); ++j) {
-                if (!(n_nodes[j].Is(CONTACT)) || !(rNode.Is(CONTACT))){
+                if ( (n_nodes[j].Is(CONTACT) && rNode.Is(CONTACT))
+                    || (!rNode.Is(CONTACT) && !n_nodes[j].Is(CONTACT)) ){
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp]:
```suggestion
                if (n_nodes[j].Is(CONTACT) == rNode.Is(CONTACT)){
```
Aren't your changes equivalent to this?

**Review by rubenzorrilla — DISMISSED** [review]
Only a minor comment. If you want to apply the changes, just let me know and I'll approve it again.

File: applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp (hunk 1 above)
Comment (mrhashemi) [reply applications/FluidDynamicsApplication/custom_processes/distance_smoothing_process.cpp]: Seems the same.

**Comment by mrhashemi** [comment]
> Only a minor comment. If you want to apply the changes, just let me know and I'll approve it again.

Thank you for the comment, it is now applied. Please re-approve.

**Review by rubenzorrilla — COMMENTED** [review]
Thanks!!

**Comment by mrhashemi** [comment]
@rubenzorrilla can you please re-approve?

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla can you please re-approve?

I thought I already approved with my last comment. Sorry.

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**